subprojects/launcher/src/main/java/org/gradle/tooling/internal/provider/StartParamsValidatingActionExecuter.java

```diff
@@ -60,6 +62,12 @@ public Object execute(BuildAction action, BuildRequestContext requestContext, Bu
                 throw new IllegalArgumentException(String.format("The specified settings file '%s' is not a file.", startParameter.getSettingsFile()));
             }
         }
+        if (startParameter.isConfigureOnDemand()) {
```

Comment:
Still, this is a conflict with intention of #2890  . Personally, I prefer #2890 's way, putting all deprecation message together to avoid writing `nagUserWith` again and again.

Reply:
It only printed them when activated through the CLI and it printed them too early.

We can open a follow-up issue to truly centralize this. That solution would have to
- work with CLI and Gradle properties
- collect messages and only log them after the logging system is active

Reply:
Missing `@SuppressWarnings("deprecation")` on this method.

Reply:
With these questions, we should open a new issue to address it. Would it be a good idea if we put each `nagUserWith` in each `BuildOption`'s `applyTo` method?

Reply:
Possibly, I haven't looked deeply at it. Let's just open an issue describing the problem and we can figure out the implementation later.

Reply:
Sure.